src/Akka.Streams.Kafka/Messages/GroupTopicPartitionOffset.cs

```diff
@@ -43,111 +43,20 @@ public GroupTopicPartitionOffset(GroupTopicPartition groupTopicPartition, Offset
     }
     
     /// <summary>
     /// Group, topic and partition info
     /// </summary>
-    public sealed class GroupTopicPartition : IEquatable<GroupTopicPartition>
-    {
-        public GroupTopicPartition(string groupId, string topic, int partition)
-        {
-            GroupId = groupId;
-            Topic = topic;
-            Partition = partition;
-        }
-        
-        /// <summary>
-        /// Consumer's group Id
-        /// </summary>
-        public string GroupId { get; }
-        /// <summary>
-        /// Topic
-        /// </summary>
-        public string Topic { get; }
-        /// <summary>
-        /// Partition
-        /// </summary>
-        public int Partition { get; }
-
-        public bool Equals(GroupTopicPartition? other)
-        {
-            if (ReferenceEquals(null, other)) return false;
-            if (ReferenceEquals(this, other)) return true;
-            return GroupId == other.GroupId && Topic == other.Topic && Partition == other.Partition;
-        }
-
-        public override bool Equals(object? obj) => ReferenceEquals(this, obj) || obj is GroupTopicPartition other && Equals(other);
-
-        public override int GetHashCode()
-        {
-            unchecked
-            {
-                var hashCode = (GroupId != null ? GroupId.GetHashCode() : 0);
-                hashCode = (hashCode * 397) ^ (Topic != null ? Topic.GetHashCode() : 0);
-                hashCode = (hashCode * 397) ^ Partition;
-                return hashCode;
-            }
-        }
-    }
-
-    public sealed class OffsetAndMetadata : IEquatable<OffsetAndMetadata>
+    public sealed record GroupTopicPartition(string GroupId, string Topic, int Partition)
     {
-        public OffsetAndMetadata(Offset offset, string metadata)
-        {
-            Offset = offset;
-            Metadata = metadata;
-        }
-
-        /// <summary>
-        /// Kafka partition offset value
-        /// </summary>
-        public Offset Offset { get; }
-        /// <summary>
-        /// Metadata
-        /// </summary>
-        public string Metadata { get; }
-
-        public bool Equals(OffsetAndMetadata? other)
-        {
-            if (ReferenceEquals(null, other)) return false;
-            if (ReferenceEquals(this, other)) return true;
-            return Offset.Equals(other.Offset) && Metadata == other.Metadata;
-        }
-
-        public override bool Equals(object? obj) => ReferenceEquals(this, obj) || obj is OffsetAndMetadata other && Equals(other);
-
-        public override int GetHashCode()
-        {
-            unchecked
-            {
-                return (Offset.GetHashCode() * 397) ^ (Metadata != null ? Metadata.GetHashCode() : 0);
-            }
-        }
+        public TopicPartition TopicPartition { get; } = new(Topic, Partition);
     }
+    
+    public sealed record OffsetAndMetadata(Offset Offset, string Metadata);
```

Comment:
We should really remove `OffsetAndMetadata` because we don't even really use `Metadata` at all - this is something that got ported from the JVM without understanding what it meant. This API wasn't an Akka thing, it was a Confluent.Kafka for Java thing. 

In the .NET client we just use `TopicPartitionOffset` instead - so deleting this in the long run would be a good idea / would reduce waste. I tried doing this on this PR and it just ballooned the size of an already large PR even larger, so better to make this its own project.